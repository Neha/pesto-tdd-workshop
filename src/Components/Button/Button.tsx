import React from 'react';
import { BUTTON } from '../../Constants/Constants';
interface props{
    click() : void
}

const Button: React.FC<props> = ({click}) => {
    return(
        <button 
            type="submit" 
            className="notActive" 
            onClick={() => click()}>{BUTTON.REFRESH}</button>
    )
}

export default Button;
