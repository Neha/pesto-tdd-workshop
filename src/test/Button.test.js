import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../Components/Button/Button';

test('renders Button component', () => {
    render(<Button />);
})

test('should have label Refresh', () => {
    render(<Button />);
    const buttonLabel = screen.getByRole('button',{ name: 'Refresh'});
    expect(buttonLabel).toBeInTheDocument();
})

test('should have type submit', () => {
    render(<Button />);
    const buttonLabel = screen.getByRole('button',{ name: 'Refresh'});
    expect(buttonLabel).toHaveAttribute('type','submit');
})

test('should have click event', () => {
    const handleClick = jest.fn();
    render(<Button click={handleClick}/>);
    const buttonLabel = screen.getByRole('button',{ name: 'Refresh'});
    fireEvent.click(buttonLabel);
    expect(handleClick).toHaveBeenCalledTimes(1);
})
test('Should match snapshot', () => {
    const handleClick = jest.fn();
    const {container} = render(<Button click={handleClick}/>);
    expect(container).toMatchSnapshot()
  });
