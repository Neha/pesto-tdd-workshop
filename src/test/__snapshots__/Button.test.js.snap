// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <button
    class="notActive"
    type="submit"
  >
    Refresh
  </button>
</div>
`;
